src/rulesets/LinearAlgebra/structured.jl

```diff
@@ -274,3 +274,24 @@ function rrule(::typeof(tril), A::AbstractMatrix)
     end
     return tril(A), tril_pullback
 end
+
+_diag_view(X) = view(X, diagind(X))
+_diag_view(X::Diagonal) = parent(X)  #Diagonal wraps a Vector of just Diagonal elements
+
+function rrule(::typeof(det), X::Union{Diagonal, AbstractTriangular})
+    y = det(X)
+    s = y ./ _diag_view(X)
```

Comment:
Why is this declared out here instead of in the pullback?

Reply:
To avoid closing over `X`
so that `X` can be garbage collected.
`s` is much smaller than `X`

Reply:
I am hoping opaque closures will do a lot of this for us automatically.
Not sure though